### PR TITLE
Support ARM (aarch64/arm64) architectures

### PR DESCRIPTION
**Pull request by jadeblaquiere**
Build process adapted to aarch64 build. rpm/deb packages install and work on Amazon A1 instances under Amazon Linux and Ubuntu 18.

### CODE REVIEW

File: internal/monitors/pyrunner/linux.go [hunk 1]
```diff
@@ -21,7 +23,11 @@ func procAttrs() *syscall.SysProcAttr {
 }
 
 func pythonBinaryExecutable() string {
-	return filepath.Join(os.Getenv(constants.BundleDirEnvVar), "lib64", "ld-linux-x86-64.so.2")
+	if strings.Compare(runtime.GOARCH, "arm64") == 0 {
+		return filepath.Join(os.Getenv(constants.BundleDirEnvVar), "lib", "ld-linux-aarch64.so.1")
+	} else {
```
Comment (keitwb) [inline internal/monitors/pyrunner/linux.go]: From golint:

`if block ends with a return statement, so drop this else and outdent its block`

File: internal/monitors/pyrunner/linux.go [hunk 2]
```diff
@@ -21,7 +23,11 @@ func procAttrs() *syscall.SysProcAttr {
 }
 
 func pythonBinaryExecutable() string {
-	return filepath.Join(os.Getenv(constants.BundleDirEnvVar), "lib64", "ld-linux-x86-64.so.2")
+	if strings.Compare(runtime.GOARCH, "arm64") == 0 {
```
Comment (keitwb) [inline internal/monitors/pyrunner/linux.go]: The idiomatic way of comparing strings in Go is with just regular `==`, none of that Java confusion.

File: internal/monitors/collectd/collectd.go [hunk 3]
```diff
@@ -468,6 +470,10 @@ func (cm *Manager) rerenderConf(writeHTTPPort int) error {
 
 func (cm *Manager) makeChildCommand() (*exec.Cmd, io.ReadCloser) {
 	loader := filepath.Join(cm.conf.BundleDir, "lib64/ld-linux-x86-64.so.2")
+	if strings.Compare(runtime.GOARCH, "arm64") == 0 {
```
Comment (keitwb) [inline internal/monitors/collectd/collectd.go]: Same thing about `==`.

**Comment by keitwb** [comment]
@jchengsfx can you take a look at the K8s failures?  They are consistent but I don't understand what changed in this branch to trigger them.

**Comment by jchengsfx** [comment]
> @jchengsfx can you take a look at the K8s failures? They are consistent but I don't understand what changed in this branch to trigger them.

For some reason, the global env vars that we have set for all jobs in the circleci project settings aren't getting set in the job. Need to investigate further.

File: internal/monitors/pyrunner/linux.go (hunk 1 above)
Comment (jadeblaquiere) [reply internal/monitors/pyrunner/linux.go]: done

File: internal/monitors/pyrunner/linux.go (hunk 2 above)
Comment (jadeblaquiere) [reply internal/monitors/pyrunner/linux.go]: done

File: internal/monitors/collectd/collectd.go (hunk 3 above)
Comment (jadeblaquiere) [reply internal/monitors/collectd/collectd.go]: done

**Comment by keitwb** [comment]
I do want to make sure AppVeyor passes to make sure this doesn't mess up anything on Windows.

**Comment by keitwb** [comment]
Joe, can you check the git commit user config on your machine and squash and amend all of this code to specify a more identifiable author?  The current commits are using names like 'root' and 'Default EC2 user' and it will be confusing in the git history.

**Comment by jadeblaquiere** [comment]
closing this request, will resubmit from a clean branch

**Pull request closed without merging**